lib/cartopy/tests/mpl/test_gridliner.py

```diff
@@ -432,3 +437,38 @@ def test_gridliner_formatter_kwargs():
     fig.canvas.draw()
     labels = [a.get_text() for a in gl.bottom_label_artists if a.get_visible()]
     assert labels == ['75°O', '70°O', '65°O', '60°O', '55°O', '50°O', '45°O']
+
+
+def test_gridliner_count_draws():
+    fig = plt.figure()
+    ax = fig.add_subplot(1, 1, 1, projection=ccrs.PlateCarree())
+    ax.set_global()
+    gl = ax.gridlines()
+
+    with mock.patch.object(gl, '_draw_gridliner', return_value=None) as mocked:
+        ax.get_tightbbox(renderer=None)
+        mocked.assert_called_once()
+
+    with mock.patch.object(gl, '_draw_gridliner', return_value=None) as mocked:
+        fig.draw_without_rendering()
+        mocked.assert_called_once()
+
+
+def test_gridliner_remove():
```

Comment:
I wanted a `check_figures_equal` for this one to show that adding and removing the gridliner gives the same as not adding it, but that doesn't seem to exist [yet] in pytest-mpl.  Another option would be an image test against an empty rectangle...

Reply:
Can you test against an already existing image?
https://github.com/SciTools/cartopy/blob/main/lib/cartopy/tests/mpl/baseline_images/mpl/test_mpl_integration/simple_global.png

Reply:
I can try - can you specify a different directory in the image test decorator?  Or should I just create a symlink from the gridliner baseline image directory to the target image?

Reply:
I have no idea... I was just pointing you at an empty figure to test against :)

FWIW, I think your current version with the asserts is fine.

Reply:
Hmmm.  I can report that the decorator takes a _baseline_dir_ keyword, but it doesn't appear to have any effect.

Reply:
Ah, it's possible if we modify `conftest.py`.  I've put that in a separate commit in case we decide that's not a good idea...